src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java

```diff
@@ -73,13 +77,15 @@ public CourseProjectAction(@NotNull CourseFactory courseFactory,
                              @NotNull SettingsImporter settingsImporter,
                              @NotNull ComponentInstaller.Factory installerFactory,
                              @NotNull PostponedRunnable ideRestarter,
-                             @NotNull CourseProjectActionDialogs dialogs) {
+                             @NotNull CourseProjectActionDialogs dialogs,
+                             @NotNull InstallerDialogs.Factory installerDialogsFactory) {
```

Comment:
A possibility to consider: there's a minor inconsistency between these two, so perhaps both could be factories or both could be interfaces